File: ios-native-shell-scaffold/VineSightShell/TabsFlowCoordinator.swift
import Foundation

enum NativeTabSelection: String, CaseIterable {
    case home
    case farms
    case tools
    case workers
    case settings
}

enum TabsFlowEvent {
    case select(tab: NativeTabSelection)
    case resetHome
}

enum TabsFlowCoordinator {
    static let defaultTab: NativeTabSelection = .home

    static func routeId(for tab: NativeTabSelection) -> NativeRouteId {
        switch tab {
        case .home: return .tabsHome
        case .farms: return .tabsFarms
        case .tools: return .tabsTools
        case .workers: return .tabsWorkers
        case .settings: return .tabsSettings
        }
    }

    static func resolveTab(from routeId: NativeRouteId) -> NativeTabSelection {
        switch routeId {
        case .tabsFarms: return .farms
        case .tabsTools: return .tools
        case .tabsWorkers: return .workers
        case .tabsSettings: return .settings
        default: return .home
        }
    }

    static func nextRoute(current: NativeRouteId, event: TabsFlowEvent) -> NativeRouteId {
        switch event {
        case .select(let tab):
            return routeId(for: tab)
        case .resetHome:
            return .tabsHome
        }
    }
}

File: android-native-shell-scaffold/app/src/main/java/com/vinesight/shell/TabsFlowCoordinator.kt
package com.vinesight.shell

enum class NativeTabSelection {
    HOME,
    FARMS,
    TOOLS,
    WORKERS,
    SETTINGS,
}

sealed interface TabsFlowEvent {
    data class Select(val tab: NativeTabSelection) : TabsFlowEvent
    data object ResetHome : TabsFlowEvent
}

object TabsFlowCoordinator {
    val defaultTab: NativeTabSelection = NativeTabSelection.HOME

    fun routeIdFor(tab: NativeTabSelection): NativeRouteId {
        return when (tab) {
            NativeTabSelection.HOME -> NativeRouteId.TABS_HOME
            NativeTabSelection.FARMS -> NativeRouteId.TABS_FARMS
            NativeTabSelection.TOOLS -> NativeRouteId.TABS_TOOLS
            NativeTabSelection.WORKERS -> NativeRouteId.TABS_WORKERS
            NativeTabSelection.SETTINGS -> NativeRouteId.TABS_SETTINGS
        }
    }

    fun resolveTab(routeId: NativeRouteId): NativeTabSelection {
        return when (routeId) {
            NativeRouteId.TABS_FARMS -> NativeTabSelection.FARMS
            NativeRouteId.TABS_TOOLS -> NativeTabSelection.TOOLS
            NativeRouteId.TABS_WORKERS -> NativeTabSelection.WORKERS
            NativeRouteId.TABS_SETTINGS -> NativeTabSelection.SETTINGS
            else -> NativeTabSelection.HOME
        }
    }

    fun nextRoute(current: NativeRouteId, event: TabsFlowEvent): NativeRouteId {
        return when (event) {
            is TabsFlowEvent.Select -> routeIdFor(event.tab)
            TabsFlowEvent.ResetHome -> NativeRouteId.TABS_HOME
        }
    }
}
